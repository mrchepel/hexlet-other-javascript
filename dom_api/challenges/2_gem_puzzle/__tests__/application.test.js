// import 'js-polyfills/dom';
import fs from 'fs';
import path from 'path';
import { html } from 'js-beautify';

import run from '../src/application';

const pressKey = (key) => {
  const e = new KeyboardEvent('keyup', { key });
  document.dispatchEvent(e);
};

const fixuturesPath = path.join(__dirname, '__fixtures__');
const getTree = () => html(document.body.innerHTML);

beforeAll(() => {
  const initHtml = fs.readFileSync(path.join(fixuturesPath, 'index.html')).toString();
  document.documentElement.innerHTML = initHtml;
  run((ar) => ar);
});

test('application', () => {
  expect(getTree()).toMatchSnapshot();

  pressKey('ArrowRight');
  pressKey('ArrowLeft');
  pressKey('ArrowLeft');
  expect(getTree()).toMatchSnapshot();

  pressKey('ArrowRight');
  expect(getTree()).toMatchSnapshot();

  pressKey('ArrowDown');
  pressKey('ArrowDown');
  expect(getTree()).toMatchSnapshot();

  pressKey('ArrowUp');
  pressKey('ArrowUp');
  expect(getTree()).toMatchSnapshot();
});
