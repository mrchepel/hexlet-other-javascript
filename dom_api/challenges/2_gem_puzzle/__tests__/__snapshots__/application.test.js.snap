// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`application 1`] = `
"<div class=\\"container m-3\\">
    <div class=\\"gem-puzzle\\">
        <table class=\\"table-bordered\\">
            <tbody>
                <tr>
                    <td class=\\"p-3\\">1</td>
                    <td class=\\"p-3\\">5</td>
                    <td class=\\"p-3\\">9</td>
                    <td class=\\"p-3\\">13</td>
                </tr>
                <tr>
                    <td class=\\"p-3\\">2</td>
                    <td class=\\"p-3\\">6</td>
                    <td class=\\"p-3\\">10</td>
                    <td class=\\"p-3\\">14</td>
                </tr>
                <tr>
                    <td class=\\"p-3\\">3</td>
                    <td class=\\"p-3\\">7</td>
                    <td class=\\"p-3\\">11</td>
                    <td class=\\"p-3\\">15</td>
                </tr>
                <tr>
                    <td class=\\"p-3\\">4</td>
                    <td class=\\"p-3\\">8</td>
                    <td class=\\"p-3\\">12</td>
                    <td class=\\"p-3 table-active\\"></td>
                </tr>
            </tbody>
        </table>
    </div>
</div>"
`;

exports[`application 2`] = `
"<div class=\\"container m-3\\">
    <div class=\\"gem-puzzle\\">
        <table class=\\"table-bordered\\">
            <tbody>
                <tr>
                    <td class=\\"p-3\\">1</td>
                    <td class=\\"p-3\\">5</td>
                    <td class=\\"p-3\\">9</td>
                    <td class=\\"p-3\\">13</td>
                </tr>
                <tr>
                    <td class=\\"p-3\\">2</td>
                    <td class=\\"p-3\\">6</td>
                    <td class=\\"p-3\\">10</td>
                    <td class=\\"p-3\\">14</td>
                </tr>
                <tr>
                    <td class=\\"p-3\\">3</td>
                    <td class=\\"p-3\\">7</td>
                    <td class=\\"p-3\\">11</td>
                    <td class=\\"p-3\\">15</td>
                </tr>
                <tr>
                    <td class=\\"p-3\\">4</td>
                    <td class=\\"p-3\\">8</td>
                    <td class=\\"p-3\\">12</td>
                    <td class=\\"p-3 table-active\\"></td>
                </tr>
            </tbody>
        </table>
    </div>
</div>"
`;

exports[`application 3`] = `
"<div class=\\"container m-3\\">
    <div class=\\"gem-puzzle\\">
        <table class=\\"table-bordered\\">
            <tbody>
                <tr>
                    <td class=\\"p-3\\">1</td>
                    <td class=\\"p-3\\">5</td>
                    <td class=\\"p-3\\">9</td>
                    <td class=\\"p-3\\">13</td>
                </tr>
                <tr>
                    <td class=\\"p-3\\">2</td>
                    <td class=\\"p-3\\">6</td>
                    <td class=\\"p-3\\">10</td>
                    <td class=\\"p-3\\">14</td>
                </tr>
                <tr>
                    <td class=\\"p-3\\">3</td>
                    <td class=\\"p-3\\">7</td>
                    <td class=\\"p-3\\">11</td>
                    <td class=\\"p-3\\">15</td>
                </tr>
                <tr>
                    <td class=\\"p-3\\">4</td>
                    <td class=\\"p-3\\">8</td>
                    <td class=\\"p-3 table-active\\"></td>
                    <td class=\\"p-3\\">12</td>
                </tr>
            </tbody>
        </table>
    </div>
</div>"
`;

exports[`application 4`] = `
"<div class=\\"container m-3\\">
    <div class=\\"gem-puzzle\\">
        <table class=\\"table-bordered\\">
            <tbody>
                <tr>
                    <td class=\\"p-3\\">1</td>
                    <td class=\\"p-3\\">5</td>
                    <td class=\\"p-3\\">9</td>
                    <td class=\\"p-3\\">13</td>
                </tr>
                <tr>
                    <td class=\\"p-3\\">2</td>
                    <td class=\\"p-3\\">6</td>
                    <td class=\\"p-3 table-active\\"></td>
                    <td class=\\"p-3\\">14</td>
                </tr>
                <tr>
                    <td class=\\"p-3\\">3</td>
                    <td class=\\"p-3\\">7</td>
                    <td class=\\"p-3\\">10</td>
                    <td class=\\"p-3\\">15</td>
                </tr>
                <tr>
                    <td class=\\"p-3\\">4</td>
                    <td class=\\"p-3\\">8</td>
                    <td class=\\"p-3\\">11</td>
                    <td class=\\"p-3\\">12</td>
                </tr>
            </tbody>
        </table>
    </div>
</div>"
`;

exports[`application 5`] = `
"<div class=\\"container m-3\\">
    <div class=\\"gem-puzzle\\">
        <table class=\\"table-bordered\\">
            <tbody>
                <tr>
                    <td class=\\"p-3\\">1</td>
                    <td class=\\"p-3\\">5</td>
                    <td class=\\"p-3\\">9</td>
                    <td class=\\"p-3\\">13</td>
                </tr>
                <tr>
                    <td class=\\"p-3\\">2</td>
                    <td class=\\"p-3\\">6</td>
                    <td class=\\"p-3\\">10</td>
                    <td class=\\"p-3\\">14</td>
                </tr>
                <tr>
                    <td class=\\"p-3\\">3</td>
                    <td class=\\"p-3\\">7</td>
                    <td class=\\"p-3\\">11</td>
                    <td class=\\"p-3\\">15</td>
                </tr>
                <tr>
                    <td class=\\"p-3\\">4</td>
                    <td class=\\"p-3\\">8</td>
                    <td class=\\"p-3 table-active\\"></td>
                    <td class=\\"p-3\\">12</td>
                </tr>
            </tbody>
        </table>
    </div>
</div>"
`;
